src/aggregation/Aggregator.ts

```diff
@@ -243,3 +244,116 @@ export class LongSumAggregator extends AbstractAggregator<Long> {
         output.writeLong(Long.ZERO);
     }
 }
+
+/** @internal */
+export class DistinctValuesAggregator<R> extends AbstractAggregator<Set<R>> {
+
+    classId = AggregatorConstants.DISTINCT;
+
+    readData(input: DataInput): any {
+        this.attributePath = input.readString();
+        const count = input.readInt();
+        for (let i = 0; i < count; i++) {
+            input.readObject();
+        }
```

Comment:
readData is not called on the client side for aggregators. Maybe we can make the method no-op